hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java

```
@@ -674,6 +685,624 @@ public Response getDeletedDirectorySummary() {
+   * a database-friendly format ("/volumeID/bucketID/ParentId/") by replacing names
+   * with their corresponding IDs. It simplifies database queries for FSO bucket operations.
+   *
+   * @param startPrefixPath The path to be converted.
+   * @return The objectId path as "/volumeID/bucketID/ParentId/".
+   * @throws IOException If database access fails.
+   */
+  public String convertStartPrefixPathToObjectIdPath(String startPrefixPath)
+      throws IOException {
+
+    String[] names = parseRequestPath(
+        normalizePath(startPrefixPath, BucketLayout.FILE_SYSTEM_OPTIMIZED));
+
+    // Root-Level :- Return the original path
+    if (names.length == 0) {
+      return startPrefixPath;
+    }
+
+    // Volume-Level :- Fetch the volumeID
+    String volumeName = names[0];
+    ReconUtils.validateNames(volumeName);
+    String volumeKey = omMetadataManager.getVolumeKey(volumeName);
+    long volumeId = omMetadataManager.getVolumeTable().getSkipCache(volumeKey)
+        .getObjectID();
+    if (names.length == 1) {
+      return ReconUtils.constructObjectPathWithPrefix(volumeId);
+    }
+
+    // Bucket-Level :- Fetch the bucketID
+    String bucketName = names[1];
+    ReconUtils.validateNames(bucketName);
+    String bucketKey = omMetadataManager.getBucketKey(volumeName, bucketName);
+    OmBucketInfo bucketInfo =
+        omMetadataManager.getBucketTable().getSkipCache(bucketKey);
+    long bucketId = bucketInfo.getObjectID();
+    if (names.length == 2) {
+      return ReconUtils.constructObjectPathWithPrefix(volumeId, bucketId);
+    }
+
+    // Fetch the immediate parentID which could be a directory or the bucket itself
+    BucketHandler handler =
+        getBucketHandler(reconNamespaceSummaryManager, omMetadataManager,
+            reconSCM, bucketInfo);
+    long dirObjectId = -1;
+    try {
+      dirObjectId = handler.getDirInfo(names).getObjectID();
+    } catch (Exception ioe) {
+      LOG.error("Not valid directory :{}", ioe);
```

If its not able to find valid path, should return error, instead of empty records

---

Ok done.